libraries/AP_Baro/AP_Baro_DPS280.cpp

```diff
@@ -0,0 +1,232 @@
+ */
+/*
+  DPS280 barometer driver
+ */
+
+#include "AP_Baro_DPS280.h"
+
+#include <utility>
+#include <stdio.h>
+
+extern const AP_HAL::HAL &hal;
+
+#define DPS280_REG_PRESS  0x00
+#define DPS280_REG_TEMP   0x03
+#define DPS280_REG_PCONF  0x06
+#define DPS280_REG_TCONF  0x07
+#define DPS280_REG_MCONF  0x08
+#define DPS280_REG_CREG   0x09
+#define DPS280_REG_ISTS   0x0A
+#define DPS280_REG_FSTS   0x0B
+#define DPS280_REG_RESET  0x0C
+#define DPS280_REG_PID    0x0D
+#define DPS280_REG_COEF   0x10
+#define DPS280_REG_CSRC   0x28
+
+#define DPS280_WHOAMI 0x10
+
+
+AP_Baro_DPS280::AP_Baro_DPS280(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::Device> _dev)
+    : AP_Baro_Backend(baro)
+    , dev(std::move(_dev))
+{
+}
+
+AP_Baro_Backend *AP_Baro_DPS280::probe(AP_Baro &baro,
+                                       AP_HAL::OwnPtr<AP_HAL::Device> _dev)
+{
+    if (!_dev) {
+        return nullptr;
+    }
+
+    AP_Baro_DPS280 *sensor = new AP_Baro_DPS280(baro, std::move(_dev));
+    if (!sensor || !sensor->init()) {
+        delete sensor;
+        return nullptr;
+    }
+    return sensor;
+}
+
+/*
+  handle bit width for 16 bit config registers
+ */
+void AP_Baro_DPS280::fix_config_bits16(int16_t &v, uint8_t bits) const
+{
+    if (v > int16_t((1U<<(bits-1))-1)) {
+        v = v - (1U<<bits);
+    }
+}
+
+/*
+  handle bit width for 32 bit config registers
+ */
+void AP_Baro_DPS280::fix_config_bits32(int32_t &v, uint8_t bits) const
+{
+    if (v > int32_t((1U<<(bits-1))-1)) {
+        v = v - (1U<<bits);
+    }
+}
+
+/*
+  read calibration data
+ */
+bool AP_Baro_DPS280::read_calibration(void)
+{
+    uint8_t buf[18];
+
+    if (!dev->read_registers(DPS280_REG_COEF, buf, 18)) {
+        return false;
+    }
+
+    calibration.C0  = (buf[0] << 4) + ((buf[1] >>4) & 0x0F);
+    calibration.C1  = (buf[2] + ((buf[1] & 0x0F)<<8));
+    calibration.C00 = ((buf[4]<<4) + (buf[3]<<12)) + ((buf[5]>>4) & 0x0F);
+    calibration.C10 = ((buf[5] & 0x0F)<<16) + buf[7] + (buf[6]<<8);
+    calibration.C01 = (buf[9] + (buf[8]<<8));
+    calibration.C11 = (buf[11] + (buf[10]<<8));
+    calibration.C20 = (buf[13] + (buf[12]<<8));
+    calibration.C21 = (buf[15] + (buf[14]<<8));
+    calibration.C30 = (buf[17] + (buf[16]<<8));
+
+    fix_config_bits16(calibration.C0, 12);
+    fix_config_bits16(calibration.C1, 12);
+    fix_config_bits32(calibration.C00, 20);
+    fix_config_bits32(calibration.C10, 20);
+    fix_config_bits16(calibration.C01, 16);
+    fix_config_bits16(calibration.C11, 16);
+    fix_config_bits16(calibration.C20, 16);
+    fix_config_bits16(calibration.C21, 16);
+    fix_config_bits16(calibration.C30, 16);
+
+    /* get calibration source */
+    if (!dev->read_registers(DPS280_REG_CSRC, &calibration.temp_source, 1)) {
+        return false;
+    }
+    calibration.temp_source &= 0x80;
+    
+	return true;
+}
+
+bool AP_Baro_DPS280::init()
+{
+    if (!dev | !dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
```

Comment:
I know in this case it ends up with the same result but it really should be || instead of |

Reply:
fixed, thanks